### ISSUE TITLE
V2 Model Less Precise

### ISSUE THREAD

**Issue by sashokbg** [MeetKai/functionary#78]
Hello,

I think that the Mistral v2 model is performing worse than the 1.4. I am not sure if this is related to the base model itself or due to training.

The context of my app is an assistant that uses the APIs of an activity report software in our company.

Here are some examples:

## 2.0 Model:

In this conversation I try to make the assistant infer the correct date from its context.
I even added the day of the week in the initial context but it fails.

```
[ {'content': 'Client stdout is HTML capable', 'role': 'system'},
  {'content': 'Current date is: Saturday 2023-12-16', 'role': 'system'},
  {'content': 'Locale is en-GB', 'role': 'system'},
  { 'content': "Currently connected user is 'aleksandar@company.com' Firstname "
               'Aleksandar Lastname KIRILOV',
    'role': 'system'},
  { 'content': 'This program helps user fill in their monthly activity '
               'reports. Each activity report is associated to a project that '
               'user has worked on. Activities can be reported in increments '
               'of 25% per day. A single day cannot have more than 100% of '
               'reported time - this includes activities and absences.',
    'role': 'system'},
  {'content': '    I was sick last friday', 'role': 'user'},
  { 'content': "Sure, let's add that as an absence for Friday, December 16th, "
               '2023. Please provide the percentage of time you were absent '
               'due to illness.',
    'role': 'assistant',
    'tool_calls': []},
  {'content': 'all day and yes illness', 'role': 'user'},
  { 'content': None,
    'role': 'assistant',
    'tool_calls': [ { 'function': { 'arguments': '{\n'
                                                 '  "email": '
                                                 '"aleksandar@company.com",\n'
                                                 '  "date": "2023-12-16",\n'
                                                 '  "absence_time": "100",\n'
                                                 '  "reason": "Maladie"\n'
                                                 '}',
                                    'name': 'add_absence'},
                      'id': 'call_CNeIHAyVRYjg22tJrmrK3PV6',
                      'type': 'function'}]},
  {'content': 'No no I mean last friday', 'role': 'user'},
  { 'content': 'Apologies for the confusion. Could you please provide the date '
               'of your absence?',
    'role': 'assistant',
    'tool_calls': []},
  {'content': '8th this month', 'role': 'user'},
  { 'content': None,
    'role': 'assistant',
    'tool_calls': [ { 'function': { 'arguments': '{\n'
                                                 '  "email": '
                                                 '"aleksandar@company.com",\n'
                                                 '  "date": "2023-12-08",\n'
                                                 '  "absence_time": "100",\n'
                                                 '  "reason": "Maladie"\n'
                                                 '}',
                                    'name': 'add_absence'},
                      'id': 'call_QsANdNZRWQAX29h5YwCKrrIt',
                      'type': 'function'}]}]

```

A second conversation in which things are even worse. First it thinks that today (Saturday) is actually Friday.
Then it incorrectly infers the date of "last Friday".

```
[ {'content': 'Client stdout is HTML capable', 'role': 'system'},
  {'content': 'Current date is: Saturday 2023-12-16', 'role': 'system'},
  {'content': 'Locale is en-GB', 'role': 'system'},
  { 'content': "Currently connected user is 'aleksandar@company.com' Firstname "
               'Aleksandar Lastname KIRILOV',
    'role': 'system'},
  { 'content': 'This program helps user fill in their monthly activity '
               'reports. Each activity report is associated to a project that '
               'user has worked on. Activities can be reported in increments '
               'of 25% per day. A single day cannot have more than 100% of '
               'reported time - this includes activities and absences.',
    'role': 'system'},
  {'content': '    I was sick last friday', 'role': 'user'},
  { 'content': "Sure, let's add that as an absence for Friday, December 16th, "
               '2023. Please provide the percentage of time you were absent '
               'due to illness.',
    'role': 'assistant',
    'tool_calls': []},
  {'content': 'No I mean the Friday before', 'role': 'user'},
  { 'content': "Ah, got it! Let's add that as an absence for Friday, December "
               '9th, 2023. Please provide the percentage of time you were '
               'absent due to illness.',
    'role': 'assistant',
    'tool_calls': []}]
```

## 1.4 Model:

With the old model things are a lot smoother.
It first asks me to confirm the information and I tell it to infer everything.
It first proposes friday (yesteday), then I correct it to "the friday before" and it correctly infers a function call.

```
[ {'content': 'Client stdout is HTML capable', 'role': 'system'},
  {'content': 'Current date is: 2023-12-16', 'role': 'system'},
  {'content': 'Locale is en-GB', 'role': 'system'},
  { 'content': "Currently connected user is 'aleksandar@company.com' Firstname "
               'Aleksandar Lastname KIRILOV',
    'role': 'system'},
  { 'content': 'This program helps user fill in their monthly activity '
               'reports. Each activity report is associated to a project that '
               'user has worked on. Activities can be reported in increments '
               'of 25% per day. A single day cannot have more than 100% of '
               'reported time - this includes activities and absences.',
    'role': 'system'},
  {'content': '    I was sick last friday', 'role': 'user'},
  { 'content': 'To add a sick leave for last Friday, please provide the '
               'following details:\n'
               '- Your email address\n'
               '- The date of the activity (in YYYY-MM-DD format)\n'
               '- Reason for the absence (Maladie)\n'
               '- The amount of time you were absent (in percentages, e.g., 50 '
               'for half a day or 100 for a full day)',
    'role': 'assistant'},
  {'content': 'You have all the needed information', 'role': 'user'},
  { 'content': None,
    'function_call': { 'arguments': '{\n'
                                    '  "email": "aleksandar@company.com",\n'
                                    '  "date": "2023-12-15",\n'
                                    '  "reason": "Maladie",\n'
                                    '  "absence_time": "100"\n'
                                    '}',
                       'name': 'add_absence'},
    'role': 'assistant'},
  {'content': 'No I meant the friday before', 'role': 'user'}]

{'role': 'assistant', 'content': None, 'function_call': {'name': 'add_absence', 'arguments': '{\n  "email": "aleksandar@company.com",\n  "date": "2023-12-08",\n  "reason": "Maladie",\n  "absence_time": "100"\n}'}}
```

**Comment by musabgultekin** [comment]
Thanks for reporting it. We're actively working on the issue, hopefully will get it working soon.

**Comment by musabgultekin** [comment]
Hi, we believe we've solved this issue with v2.2. If you have a chance, can you check it out?

**Comment by jeffreymeetkai** [comment]
I will be closing this issue now. Feel free to reopen if observe similar issues in v2.2!
